dev/breeze/src/airflow_breeze/utils/selective_checks.py

```diff
@@ -885,18 +882,20 @@ def _get_providers_test_types_to_run(self, split_to_individual_providers: bool =
                     include_docs=False,
                 )
         candidate_test_types: set[str] = set()
-        if affected_providers and not isinstance(affected_providers, AllProvidersSentinel):
+        if isinstance(affected_providers, AllProvidersSentinel):
+            if split_to_individual_providers:
+                for provider in get_available_packages():
+                    candidate_test_types.add(f"Providers[{provider}]")
+            else:
+                candidate_test_types.add("Providers")
+        elif affected_providers:
             if split_to_individual_providers:
                 for provider in affected_providers:
                     candidate_test_types.add(f"Providers[{provider}]")
             else:
                 candidate_test_types.add(f"Providers[{','.join(sorted(affected_providers))}]")
-        elif split_to_individual_providers:
```

Comment:
Bug number two: we missed `else` here. In effect when "split_to_individual_providers" were set to False, and "AllProvidersSentinel" was set as "affected_providers" -> candidate_test_types were empty (which lead to "skip_provider_tests=True"

I rewrote the ifs to be more logical and easier to understand.